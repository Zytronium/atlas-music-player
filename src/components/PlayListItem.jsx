import SongTitle from "./SongTitle.jsx";

export default function PlayListItem(props) {
  return (
    <button
      className={`flex w-full flex-row items-center justify-between ${props.active ? "bg-tertiary" : ""} h-10 cursor-pointer rounded-md px-1`}
    >
      <SongTitle title={props.title} artist={props.artist} size={"small"} />
      <p className="text-muted dark:text-dark-muted text-sm font-[500]">
        {props.duration}
      </p>
    </button>
  );
}
