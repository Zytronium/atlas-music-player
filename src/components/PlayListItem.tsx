import SongTitle from "./SongTitle";
import { JSX } from "react";

type Props = {
  duration: string;
  title: string;
  artist: string;
  active?: boolean;
};

export default function PlayListItem(props: Props): JSX.Element {
  return (
    <button
      className={`flex w-full flex-row items-center justify-between ${props.active ? "bg-tertiary" : ""} h-10 cursor-pointer rounded-md px-1`}
    >
      <SongTitle title={props.title} artist={props.artist} size={"small"} />
      <p className="text-muted dark:text-dark-muted text-sm font-[500]">
        {props.duration}
      </p>
    </button>
  );
}
