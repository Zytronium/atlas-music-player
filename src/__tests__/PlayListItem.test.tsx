import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayListItem from "../components/PlayListItem";

test("PlayListItem - typical props (inactive)", () => {
  const { container } = render(
    <PlayListItem
      duration={125}
      title="Typical Track"
      artist="Some Artist"
      index={0}
      setSongIndex={() => {}}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem - active state applied", () => {
  const { container } = render(
    <PlayListItem
      duration={200}
      title="Active Track"
      artist="Some Other Artist"
      index={1}
      setSongIndex={() => {}}
      active
    />,
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem - active explicitly false", () => {
  const { container } = render(
    <PlayListItem
      duration={200}
      title="Explicitly Inactive"
      artist="Yet Another Artist"
      index={5}
      setSongIndex={() => {}}
      active={false}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem - zero duration and short title", () => {
  const { container } = render(
    <PlayListItem
      duration={0}
      title="Short"
      artist="a"
      index={2}
      setSongIndex={() => {}}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem - NaN duration", () => {
  const { container } = render(
    <PlayListItem
      duration={Number.NaN as unknown as number}
      title="Weird Duration"
      artist="Edge Case"
      index={3}
      setSongIndex={() => {}}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem - duration passed as MM:SS string displays as NaN", () => {
  const { container } = render(
    <PlayListItem
      duration={"3:45" as unknown as number}
      title="String Duration"
      artist="Edge Case"
      index={6}
      setSongIndex={() => {}}
    />,
  );
  expect(container).toMatchSnapshot();
});

test("PlayListItem - long duration", () => {
  const { container } = render(
    <PlayListItem
      duration={3601}
      title="Never gonna give you up"
      artist="Rick Astley"
      index={4}
      setSongIndex={() => {}}
    />,
  );
  expect(container).toMatchSnapshot();
});
