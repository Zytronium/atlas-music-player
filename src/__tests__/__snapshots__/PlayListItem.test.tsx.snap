// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem - NaN duration 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between  h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        Weird Duration
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        Edge Case
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      ??:??
    </p>
  </button>
</div>
`;

exports[`PlayListItem - active explicitly false 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between  h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        Explicitly Inactive
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        Yet Another Artist
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      3:20
    </p>
  </button>
</div>
`;

exports[`PlayListItem - active state applied 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between bg-tertiary h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        Active Track
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        Some Other Artist
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      3:20
    </p>
  </button>
</div>
`;

exports[`PlayListItem - duration passed as MM:SS string displays as NaN 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between  h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        String Duration
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        Edge Case
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      ??:??
    </p>
  </button>
</div>
`;

exports[`PlayListItem - long duration 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between  h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        Never gonna give you up
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        Rick Astley
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      60:01
    </p>
  </button>
</div>
`;

exports[`PlayListItem - typical props (inactive) 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between  h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        Typical Track
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        Some Artist
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      2:05
    </p>
  </button>
</div>
`;

exports[`PlayListItem - zero duration and short title 1`] = `
<div>
  <button
    class="flex w-full flex-row items-center justify-between  h-10 cursor-pointer rounded-md px-1"
  >
    <div
      class="flex flex-col"
    >
      <h2
        class="text-sm font-[500] text-left"
      >
        Short
      </h2>
      <p
        class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
      >
        a
      </p>
    </div>
    <p
      class="text-muted dark:text-dark-muted text-sm font-[500]"
    >
      0:00
    </p>
  </button>
</div>
`;
